### PR TITLE
Add arp packet example

### PR DESCRIPTION
**Pull request by Dineshs91**
I've hardcoded some sample values, is it fine ?

### CODE REVIEW

**Comment by Dineshs91** [comment]
I think I at least need to get the interface from cmd line args

File: examples/arp_packet.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    ///     protocol_type: [0x08, 0x00], // Ipv4(0x0800)
+    ///     hw_addr_len: [0x06],
+    ///     proto_addr_len: [0x04],
+    ///     operation: [0x00, 0x02], // Reply(0x0002)
+    ///     sender_hw_addr: [0x28, 0xef, 0xf9, 0x5f, 0x8e, 0x2b],
+    ///     sender_proto_addr: [0xc0, 0xa8, 0x00, 0x66], // Ipv4(192.168.0.102)
+    ///     target_hw_addr: [0xff, 0xff, 0xff, 0xff, 0xff, 0xff], // Broadcast
+    ///     target_proto_addr: [0xc0, 0xa8, 0x00, 0x65], // Ipv4(192.168.0.101)
+    ///     payload: [],
+    /// }
```
Comment (mrmonday) [inline examples/arp_packet.rs]: I'd remove this block of code.

File: examples/arp_packet.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    let mut ethernet_buffer: Vec<u8> = (0..42).map(|_| 0x00).collect::<Vec<u8>>();
```
Comment (mrmonday) [inline examples/arp_packet.rs]: Usually, for allocating packets, you would use:
```rust
let mut ethernet_buffer = [0u8; 42];
```
instead, which will allocate 42 bytes of zeroes on the stack.

File: examples/arp_packet.rs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    ethernet_packet.set_payload(arp_packet.packet_mut());
+
+    tx.send_to(&ethernet_packet.to_immutable(), Some(interface));
+}
+
+fn main() {
+    let source_ip: Result<Ipv4Addr, AddrParseError> = "192.168.0.2".parse();
+    let source_mac: Result<MacAddr, ParseMacAddrErr> = "28:cf:e9:e1:43:bb".parse();
+    let target_ip: Result<Ipv4Addr, AddrParseError> = "192.168.0.3".parse();
+    let target_mac: Result<MacAddr, ParseMacAddrErr> = "28:cf:e9:e1:21:ba".parse();
```
Comment (mrmonday) [inline examples/arp_packet.rs]: You should take these values as command line parameters instead.

You could use defaults from the network interface, if you like.

**Comment by mrmonday** [comment]
I've left a few comments, this looks good overall. With them addressed, I'll be happy to merge this. Thank you for doing this!

**Comment by Dineshs91** [comment]
I've addressed your comments. Let me know if I need to change anything. 

Thanks.

**Comment by mrmonday** [comment]
I'll take another look through tomorrow when I have some time, sorry I can't do so now!

**Comment by mrmonday** [comment]
Sorry for the slow turnaround on this!

Looking good, thanks!

**Pull request merged**